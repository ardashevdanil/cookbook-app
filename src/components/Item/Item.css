.Item {
	width: 312px;
	padding: 20px calc( (100% - 312px * 3)/6 );
}

.Item:hover {
  cursor: pointer;
  color: #911026;
}

.Item>img {
	width: 100%;
}

.Item__name {
	font-size: 16.5px;
	margin-top: 9px;
}

.Item__stats {
	font-size: 14px;
	margin-top: 7.5px;
}

.Item svg {
	height: 11px
}

@media only screen and (max-width: 1120px) {
  .Item {
    width: 100%;
    padding-top: 0;
  }

  .Item>img {
    height: 64.7vw;
  }

  .Item__name {
    font-size: 5.2vw;
    margin-top: 2.8vw;
  }

  .Item__stats {
    font-size: 4.4vw;
    margin-top: 2.3vw;
  }

  .Item svg {
    height: 3.4vw;
  }

  .Item__description {
    padding: 0 6.9vw;
  }
}