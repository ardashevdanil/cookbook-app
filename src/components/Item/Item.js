import PropTypes from 'prop-types';
import React from 'react';
import './Item.css';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
	faClock,
	faComment,
	faEye
} from '@fortawesome/free-regular-svg-icons';

function Item(props) {

	return (
		<div className='Item'>
			<img
				alt={ props.content.name}
				src={ props.content.img }
			/>
			<div className='Item__description'>
				<div className='Item__name'>
					{ props.content.name }
				</div>
				<div className='Item__stats'>
					<FontAwesomeIcon icon={ faClock } />
					{ props.content.cookingTime } mins |{' '}
					<FontAwesomeIcon icon={ faComment } />
					{ props.content.comments }
					<FontAwesomeIcon icon={ faEye } />
					{ props.content.watched }
				</div>
			</div>
		</div>
	);
}

Item.propTypes = {
	content: PropTypes.shape({
		img: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired,
		comments: PropTypes.number.isRequired,
		cookingTime: PropTypes.number.isRequired,
		watched: PropTypes.number.isRequired,
	})
}

export default Item;