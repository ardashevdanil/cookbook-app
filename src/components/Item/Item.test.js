import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import Item from './Item';

Enzyme.configure({ adapter: new Adapter() });

describe('test Item component', () => {
  const pushCart = jest.fn();
	const component = Enzyme.shallow(
		<Item
      content={{
        img: '',
        name: 'test',
      }}
    />
	);

	test('renders correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});

  /*
  test('calls pushCart after click on the cart button', () => {
    component.find('div').at(6).simulate('click');

    expect(pushCart.mock.calls.length).toBe(1);
  })
  */
})