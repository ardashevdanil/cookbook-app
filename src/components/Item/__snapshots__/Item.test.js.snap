// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Item component renders correctly 1`] = `
"<div className=\\"Item\\">
  <img alt=\\"test\\" src=\\"\\" />
  <div className=\\"Item__description\\">
    <div className=\\"Item__name\\">
      test
    </div>
    <div className=\\"Item__stats\\">
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} transform={{...}} />
       mins |
       
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} transform={{...}} />
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} transform={{...}} />
    </div>
  </div>
</div>"
`;
